model.py

```diff
@@ -1859,6 +1861,45 @@ def type_and_identifier_for_urn(cls, identifier_string):
         return (type, identifier_string)
 
     @classmethod
+    def parse_urns(cls, _db, identifier_strings):
```

Comment:
I see how this method saves time when you give it a bunch of identifiers that are already in the catalog. Is that all, or does it also save time when the identifiers are new?

Reply:
It doesn't save time when identifiers are new, but it could. Right now `Identifier.for_foreign_id` does some final evaluations on the identifier, but I'll refactor to apply them separately.